### PR TITLE
trace API: fix missing/inconsistent doc of category items

### PR DESCRIPTION
**Pull request by DDvO**
Make `OSSL_trace_set_channel.pod` and `openssl.pod` consistent again with `trace.h`.

### CODE REVIEW

File: doc/man1/openssl.pod [hunk 1]
```diff
@@ -747,28 +751,48 @@ of generated for each change.
...
+Traces encoder operations.
+
+=item B<REF_COUNT>
+
+Traces decrementing certain ASN.1 structure references.
 
 =item B<HTTP>
 
-HTTP client diagnostics
+HTTP client diagnostics.
```
Comment (levitte) [inline doc/man1/openssl.pod:795]: *Traces the* HTTP client diagnostics.

File: doc/man3/OSSL_trace_set_channel.pod [hunk 2]
```diff
@@ -180,9 +184,29 @@ point during evaluation.
...
+
+Traces encoder operations.
+
+=item B<OSSL_TRACE_CATEGORY_ENGINE_REF_COUNT>
+
+Traces decrementing certain ASN.1 structure references.
+
+=item B<OSSL_TRACE_CATEGORY_HTTP>
+
+HTTP client diagnostics.
```
Comment (levitte) [inline doc/man3/OSSL_trace_set_channel.pod:209]: *Traces the* HTTP client diagnostics.

File: doc/man1/openssl.pod (hunk 1 above)
Comment (DDvO) [reply doc/man1/openssl.pod]: Changed to: `Traces the HTTP client, such as message headers being sent and received.`

File: doc/man3/OSSL_trace_set_channel.pod (hunk 2 above)
Comment (DDvO) [reply doc/man3/OSSL_trace_set_channel.pod]: Changed to: `Traces the HTTP client, such as message headers being sent and received.`

**Comment by DDvO** [comment]
Squashed and merged to 3.0 and master - thanks @levitte and @t8m

**Pull request closed without merging**